internal/pkg/deploy/cloudformation/stack/env_integration_test.go

```diff
@@ -318,3 +314,125 @@ network:
 		})
 	}
 }
+
+func TestEnvStack_Regression(t *testing.T) {
+	testCases := map[string]struct {
+		originalManifest *deploy.CreateEnvironmentInput
+		newManifest      *deploy.CreateEnvironmentInput
+	}{
+		"produce the same template after migrating environment manifest to the new field hierarchy": {
```

Comment:
This is such a good integ test ❤️ !

Just a tiny nit: can we make the test case name more specific?
```suggestion
		"should produce the same template after migrating load balancer ingress fields": {
```